### PR TITLE
add baseline localgroups plugin

### PR DESCRIPTION
**Pull request by ZeArioch**
I _may_ have twisted RegRippy's logic to retrieve information both from the SOFTWARE and the SAM hive at the same time. Would appreciate your input on this!

### CODE REVIEW

File: regrippy/plugins/localgroups.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        """Gets user SIDs from the SAM hive"""
+        machine_sid = self.machine_sid()
+        key_accounts = self.open_key("SAM\\Domains\\Account\\Users\\Names")
+        if not key_accounts:
+            return
+        for subkey in key_accounts.subkeys():
+            # User SIDs are built by appending their Relative ID to the machine SID.
+            # Relative IDs are stored inside Names' subkeys; each has a default value where the value *type* holds the
+            # integer value of the RID.
+            sid = "{0}-{1}".format(machine_sid, subkey.value("").value_type())
```
Comment (simsor) [inline regrippy/plugins/localgroups.py]: Small nitpick: could you use `.value("(default)")` when getting the default value of a key? I find it more explicit.

**Comment by simsor** [comment]
Ahah, this is indeed a use case where the plugin system works *against* the developer...

The way you handled it seems decent enough, I don't really see any other way to do it anyway. `__REGHIVE__` being a list, and thus being ordered, it seems reasonable to expect the plugin to be called in the order specified.

Also, could you please add a test file for this new plugin to your PR? I'll be more than happy to merge afterwards 😄

File: regrippy/plugins/localgroups.py (hunk 1 above)
Comment (ZeArioch) [reply regrippy/plugins/localgroups.py]: Fixed in 641000a

**Comment by ZeArioch** [comment]
> Also, could you please add a test file for this new plugin to your PR? I'll be more than happy to merge afterwards 😄

Done in f523427!

**Pull request merged**